lib/ansible/modules/cloud/azure/azure_rm_acs.py

```diff
@@ -393,12 +401,24 @@ def create_master_profile_dict(masterprofile):
     :param: masterprofile: ContainerServiceMasterProfile with the Azure callback object
     :return: dict with the state on Azure
     '''
-    results = dict(
+    return dict(
         count=masterprofile.count,
         fqdn=masterprofile.fqdn,
+        vm_size=masterprofile.vm_size,
         dns_prefix=masterprofile.dns_prefix
     )
-    return results
+
+
+def create_service_principal_profile_dict(serviceprincipalprofile):
+    '''
+    Helper method to deserialize a ContainerServiceServicePrincipalProfile to a dict
+    :param: serviceprincipalprofile: ContainerServiceServicePrincipalProfile with the Azure callback object
+    :return: dict with the state on Azure
+    '''
+    return dict(
+        client_id=serviceprincipalprofile.client_id,
+        secret=serviceprincipalprofile.secret
```

Comment:
I wouldn't return this value, if it is a secret then we don't want to expose it.

Reply:
You're quite right on this. Actually the secret in response json is empty. Let me remove the secret key from the dict.